packages/jellyfish-api-core/__tests__/category/loan/auctionBid.test.ts

```diff
@@ -0,0 +1,350 @@
+    })
+
+    it('should not auctionBid as bid token does not match auction one', async () => {
+      const promise = tGroup.get(1).rpc.loan.auctionBid({
+        vaultId: vaultId,
+        index: 0,
+        from: node1ColAddr,
+        amount: '10000@DFI'
+      })
+      await expect(promise).rejects.toThrow(RpcApiError)
+      await expect(promise).rejects.toThrow('Bid token does not match auction one')
+    })
+
+    it('should not auctionBid as insufficient fund', async () => {
+      const node1ColAcc = await tGroup.get(1).rpc.account.getAccount(node1ColAddr)
+      const tslaAcc = node1ColAcc.find((amt: string) => amt.split('@')[1] === 'TSLA')
+      const tslaAmt = Number(tslaAcc?.split('@')[0])
+      expect(Number(tslaAmt)).toBeLessThan(30000)
+
+      const promise = tGroup.get(1).rpc.loan.auctionBid({
+        vaultId: vaultId,
+        index: 0,
+        from: node1ColAddr,
+        amount: '30000@TSLA'
+      })
+      await expect(promise).rejects.toThrow(RpcApiError)
+      await expect(promise).rejects.toThrow(`amount ${tslaAmt} is less than 30000.00000000`)
+    })
+
+    it('should not auctionBid as first bid should include liquidation penalty of 5%', async () => {
+      const promise = tGroup.get(1).rpc.loan.auctionBid({
+        vaultId: vaultId,
+        index: 0,
+        from: node1ColAddr,
+        amount: '500@TSLA' // (500 * 5%) + 500 = 525, should not less than 525
+      })
+      await expect(promise).rejects.toThrow(RpcApiError)
+      await expect(promise).rejects.toThrow('First bid should include liquidation penalty of 5%')
+    })
+
+    it('should auctionBid', async () => {
+      const node1ColAccBefore = await tGroup.get(1).rpc.account.getAccount(node1ColAddr)
+      const node1BeforeTSLAAcc = node1ColAccBefore.length > 0
+        ? node1ColAccBefore.find((amt: string) => amt.split('@')[1] === 'TSLA')
+        : undefined
+      const node1BeforeTSLAAmt = node1BeforeTSLAAcc !== undefined ? Number(node1BeforeTSLAAcc.split('@')[0]) : 0
+
+      const txid = await tGroup.get(1).rpc.loan.auctionBid({
+        vaultId: vaultId,
+        index: 0,
+        from: node1ColAddr,
+        amount: '525@TSLA' // (500 * 5%) + 500 = 525, min first bid is 525
+      })
+      expect(typeof txid).toStrictEqual('string')
+      await tGroup.get(1).container.generate(1)
+      await tGroup.waitForSync()
+
+      const auctionsBefore = await tGroup.get(1).container.call('listauctions')
+
+      const node1ColAccAfter = await tGroup.get(1).rpc.account.getAccount(node1ColAddr)
+      const node1AfterTSLAAcc = node1ColAccAfter.length > 0
+        ? node1ColAccAfter.find((amt: string) => amt.split('@')[1] === 'TSLA')
+        : undefined
+      const node1AfterTSLAAmt = node1AfterTSLAAcc !== undefined ? Number(node1AfterTSLAAcc.split('@')[0]) : 0
+      expect(node1BeforeTSLAAmt - node1AfterTSLAAmt).toStrictEqual(525)
+
+      // end the auction and node1 get the auction bid
+      await tGroup.get(0).generate(36)
```

Comment:
should `waitForSync()` after this?

Reply:
you're right... it works without sync which strange me.. hold on...